src/publishProject/projectUtils.ts

```diff
@@ -140,3 +159,41 @@ export function getPublishServerName(target: string) {
 export function validateSqlServerPortNumber(port: number): boolean {
     return Number.isInteger(port) && port >= 1 && port <= constants.MAX_PORT_NUMBER;
 }
+
+/**
+ * Retrieves and filters SQL Server container tags based on project target version.
+ * Used by Publish Project dialog to provide granular tag selection.
+ *
+ * @param targetVersion - The SQL Server version (e.g., "160" for SQL Server 2022)
+ * @returns Sorted array of tags filtered by version from deployment UI versions
+ */
+export async function getSqlServerContainerTagsForTargetVersion(
+    targetVersion?: string,
+): Promise<FormItemOptions[]> {
+    try {
+        // Get the deployment UI versions first
+        const deploymentVersions = await getSqlServerContainerVersions();
+        if (!deploymentVersions || deploymentVersions.length === 0) {
+            return [];
+        }
+
+        // Determine minimum year based on target version
+        let minYear = 2017; // Default to SQL Server 2017
+        if (targetVersion) {
+            const versionNum = parseInt(targetVersion, 10);
+            // Find the corresponding year, or default to 2017 if not found
+            minYear = DSP_VERSION_TO_YEAR.get(versionNum) ?? 2017;
```

Comment:
This should refer to a `const DEFAULT_YEAR` instead of hardcoded value.  It'll be easier to change without missing a spot.